src/dataregistry/registrar/dataset.py

```diff
@@ -27,10 +30,18 @@
 
 
 class DatasetTable(BaseTable):
-    def __init__(self, db_connection, root_dir, owner, owner_type, execution_table):
+    def __init__(
```

Comment:
Since you've added annotations parameters here, for consistency it would be better to do it for all functions in this file.